cli/cmd/status/status.go

```diff
@@ -216,24 +214,39 @@ func validEvent(events []release.HookEvent) bool {
 	return false
 }
 
-// checkConsulServers uses the Kubernetes list function to report if the consul servers are healthy.
-func (c *Command) checkConsulServers(namespace string) (string, error) {
-	servers, err := c.kubernetes.AppsV1().StatefulSets(namespace).List(c.Ctx,
-		metav1.ListOptions{LabelSelector: "app=consul,chart=consul-helm,component=server"})
+// checkConsulAgents prints the status of Consul servers and clients if they
+// are expected to be found in the Kubernetes cluster. It does not check for
+// server status if they are not running within the Kubernetes cluster.
+func (c *Command) checkConsulAgents(namespace string) error {
+	// Check clients (TODO this can be removed when more users are using Agentless Consul - t-eckert 7 Oct 22)
+	clients, err := c.kubernetes.AppsV1().DaemonSets(namespace).List(c.Ctx, metav1.ListOptions{LabelSelector: "app=consul,chart=consul-helm"})
 	if err != nil {
-		return "", err
-	} else if len(servers.Items) == 0 {
-		return "", errors.New("no server stateful set found")
-	} else if len(servers.Items) > 1 {
-		return "", errors.New("found multiple server stateful sets")
+		return err
+	}
+	if len(clients.Items) != 0 {
```

Comment:
minor nit.  you could make this a shared func for this and below and pass in:
- agentType string (servers vs clients.  also used for string replacement in the output)
- desiredAgents int
- readyAgents int

Reply:
I appreciate this feedback! I did think about doing that to "DRY" out this function. My thinking with the current way it is written is that one day we will nix the clients check and I don't want an extra function floating around. I think it's nice to just delete the specific lines and not have to change anything else. Once those lines are gone, there won't be any more deduping.

What do you think of that? Am I overthinking it?